### DISCUSSION TITLE
Prepared statement needs to be re-prepared with sql views

### DISCUSSION THREAD

**Discussion by jemink** [prisma/prisma#7924] [Q&A] [answered]
I created sql views with two tables called Draft

```sql
CREATE VIEW `Draft` AS
    SELECT company.address as address, company.id as id
    FROM `company`, `admin`
    WHERE `company`.`company_id` = `admin`.`id`;
```
Then added Draft model inside the schema.prisma like below
```prisma
model Draft {
  id      Int    @unique
  address String
}
```
But when I run this query in playground I am getting below error with node js and with playground

const data = await ctx.prisma.draft.findMany({})

ERROR:-
```
"message": "\nInvalid `prisma.draft.findMany()` invocation:\n\n\n  Error occurred during query execution:\nConnectorError(ConnectorError { user_facing_error: None, kind: QueryError(Server(ServerError { code: 1615, message: \"Prepared statement needs to be re-prepared\", state: \"HY000\" })) })
```

**Comment by ryands17** [comment, marked as answer]
@jemink 👋 

Can you share your `Company` and `Address` models as well?

**Reply by jemink** [reply]
I also tried this docs https://www.prisma.io/docs/guides/database/advanced-database-tasks/sql-views-mysql

Created below error
```sql
CREATE TABLE `SqlViews`.`User` (
  `id` INT AUTO_INCREMENT PRIMARY KEY,
  `name` VARCHAR(256),
  `email` VARCHAR(256) UNIQUE
);

CREATE TABLE `SqlViews`.`Post` (
  `id` INT AUTO_INCREMENT PRIMARY KEY,
  `title` VARCHAR(256),
  `content` VARCHAR(256),
  `published` BOOLEAN,
  `authorId` INT,
  CONSTRAINT `author` FOREIGN KEY (`authorId`) REFERENCES `User`(`id`)
);
```

And created view
```sql
CREATE VIEW Draft AS
    SELECT published, title, email, Post.id
    FROM Post, User
    WHERE published = false AND Post.authorId = User.id;
```

Then added model in schema.prisma
```prisma
model Post {
  authorId  Int?
  content   String?
  id        Int     @default(autoincrement()) @id
  published Boolean @default(false)
  title     String
  User      User?   @relation(fields: [authorId], references: [id])
}

model User {
  email String  @unique
  id    Int     @default(autoincrement()) @id
  name  String?
  Post  Post[]
}

 model Draft {
   title     String
   id        Int     @unique
   email     String
   published Boolean
  }
```
![image](https://user-images.githubusercontent.com/28702189/123789211-8afa1180-d8fa-11eb-85ab-34103c3a8712.png)

**Reply by ryands17** [reply]
Works fine for me. What version of `prisma`, `@prisma/client`, and MySQL are you using?

**Reply by jemink** [reply]
"@prisma/client": "2.23.0",
"prisma": "2.23.0",

**Reply by jemink** [reply]
mariadb: 10.4.18-MariaDB-1:10.4.18+maria~focal

**Reply by janpio** [reply]
Now an issue at https://github.com/prisma/prisma/issues/7906

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
